File: src/main/kotlin/com/tutu/wrath/modules/income/models/Income.kt
package com.tutu.wrath.modules.income.models

import com.tutu.wrath.anger.tables.RowValue
import com.tutu.wrath.modules.accounting.models.Account
import com.tutu.wrath.modules.accounting.models.Frequency
import com.tutu.wrath.util.Chrono
import com.tutu.wrath.util.Money

data class Income(
    val name: String,
    val amount: Money,
    val origin: Account,
    val data: Chrono,
    val frequency: Frequency
) : RowValue {

    override fun getRowValue(id: String): String {
        return when(id) {
            "name" -> name
            "amount" -> amount.display()
            "origin" -> origin.getName()
            else -> "-"
        }
    }
}

File: src/modules/income/models/income.rs
use crate::anger::table::row::RowValue;
use crate::util::money::Money;
use crate::modules::accounting::{Account, Frequency};
use chrono::{Local, DateTime};

//TODO RETIRAR PUB QUANDO TERMINAR COMPONENTE
#[derive(Debug, Clone, PartialEq, Hash)]
pub struct Income<A: Account> {
    pub name: String,
    pub amount: Money,
    pub origin: A,
    pub date: DateTime<Local>,
    pub frequency: Frequency
}


impl<A: Account> RowValue for Income<A> {
    fn key(&self)-> String {
        format!("{}-{}", self.name, self.amount.0)
    }
}